### PR TITLE
Added benchmarks for examples/update

### LINKED ISSUE #501: examples/update benchmarks
Re-write the example at examples/update in Rust/Motoko and generate corresponding benchmarks.

- [x] Rust
- [x] Motoko
- [x] Azle
- [x] Benchmarks

### PR DESCRIPTION
**Pull request by dansteren**
This adds a Rust and Motoko implementation of this example and benchmarks for all 3 implementations.

Resolves #501

### CODE REVIEW

File: examples/update/canisters/azle/index.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    perf_result = {
+        wasm_body_only: end - start,
+        wasm_including_prelude: ic.performance_counter(0)
+    };
+}
+//#endregion
+
+let currentMessage: string = '';
+
+export function query(): Query<string> {
```
Comment (lastmjs) [inline examples/update/canisters/azle/index.ts:25]: The Azle and Motoko names for this function should be the same, perhaps something like `get_current_message` would be better than `query1`?

File: examples/update/canisters/motoko/main.mo [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+      wasm_body_only = end - start;
+      wasm_including_prelude = Prim.performanceCounter(0);
+    };
+  };
+  //#endregion
+
+  var currentMessage = "";
+
+  // query is a reserved word and therefore cannot be a function name
+  public query func query1() : async Text {
```
Comment (lastmjs) [inline examples/update/canisters/motoko/main.mo:27]: The Azle and Motoko names for this function should be the same, perhaps something like `get_current_message` would be better than `query1`?

File: examples/update/canisters/rust/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,51 @@
+use std::cell::RefCell;
+
+thread_local! {
+    static CURRENT_MESSAGE: RefCell<String> = RefCell::new(String::from(""));
```
Comment (lastmjs) [inline examples/update/canisters/rust/src/lib.rs:4]: Rename to `CURRENT_MESSAGE_REF_CELL`

Comment (lastmjs) [inline examples/update/canisters/rust/src/lib.rs:4]: Our convention is "".to_string()

File: examples/update/canisters/rust/src/lib.rs [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+            wasm_body_only: end - start,
+            wasm_including_prelude: ic_cdk::api::call::performance_counter(0),
+        })
+    });
+}
+//#endregion
+
+#[ic_cdk_macros::query]
+fn query() -> String {
+    CURRENT_MESSAGE.with(|current_message| current_message.borrow().clone())
```
Comment (lastmjs) [inline examples/update/canisters/rust/src/lib.rs:38]: Rename to `CURRENT_MESSAGE_REF_CELL` and `current_message_ref_cell`

File: examples/update/canisters/rust/src/lib.rs [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+}
+
+#[ic_cdk_macros::update]
+fn update(message: String) -> () {
+    let perf_start = ic_cdk::api::call::performance_counter(0);
+
+    CURRENT_MESSAGE.with(|current_message| {
```
Comment (lastmjs) [inline examples/update/canisters/rust/src/lib.rs:45]: Rename to `CURRENT_MESSAGE_REF_CELL` and `current_message_ref_cell`

File: examples/update/dfx.json [hunk 6]
```diff
@@ -1,14 +1,28 @@
...
+            "ts": "canisters/azle/index.ts",
+            "candid": "canisters/azle/index.did",
+            "wasm": "target/wasm32-unknown-unknown/release/azle.wasm",
             "declarations": {
-                "output": "test/dfx_generated/update"
+                "output": "dfx_generated/azle"
+            }
+        },
+        "motoko": {
+            "main": "canisters/motoko/main.mo",
```
Comment (lastmjs) [inline examples/update/dfx.json:15]: Let's keep the dfx.json files using the same conventions, the motoko should have type: motoko, and the rust should have type: rust first, then package, then candid

File: examples/update/package.json [hunk 7]
```diff
@@ -1,5 +1,6 @@
 {
     "scripts": {
+        "benchmark": "ts-node --transpile-only --ignore=false benchmark/benchmark.ts",
```
Comment (lastmjs) [inline examples/update/package.json:3]: The convention has been to put benchmark below test, if we want to change that we should do it across all of the examples

**Review by lastmjs — CHANGES_REQUESTED** [review]

File: examples/update/canisters/rust/src/lib.old.did [hunk 8]
```diff
@@ -0,0 +1,10 @@
+type PerfResult = record {
+    "wasm_body_only": nat64;
+    "wasm_including_prelude": nat64;
+};
+
+service: {
+    "get_perf_result": () -> (opt PerfResult) query;
+    "get_message": () -> (text) query;
+    "set_message": (text) -> ();
+}
```
Comment (dansteren) [inline examples/update/canisters/rust/src/lib.old.did:10]: :man_facepalming: Need to remove this

File: examples/update/benchmarks.md [hunk 9]
```diff
@@ -0,0 +1,33 @@
+# Azle/Motoko/Rust Benchmarks
```
Comment (lastmjs) [inline examples/update/benchmarks.md:1]: Let's run this again with USD Cost estimates

File: examples/update/canisters/rust/src/lib.old.did (hunk 8 above)
Comment (lastmjs) [reply examples/update/canisters/rust/src/lib.old.did:10]: Remove this

**Review by lastmjs — CHANGES_REQUESTED** [review]

**Pull request merged**